drivers/display/display_ili9806e_dsi.c

```diff
@@ -0,0 +1,440 @@
+	{.reg = 0xff, .cmd_len = 5, .cmd = {0xff, 0x98, 0x06, 0x04, 0x07}},
+	/* VREG1/2OUT ENABLE */
+	{.reg = 0x18, .cmd_len = 1, .cmd = {0x1d}},
+	{.reg = 0x26, .cmd_len = 1, .cmd = {0xb2}},
+	{.reg = 0x02, .cmd_len = 1, .cmd = {0x77}},
+	{.reg = 0xe1, .cmd_len = 1, .cmd = {0x79}},
+	{.reg = 0x17, .cmd_len = 1, .cmd = {0x22}},
+	/* Change to Page 0 CMD for Normal command */
+	{.reg = 0xff, .cmd_len = 5, .cmd = {0xff, 0x98, 0x06, 0x04, 0x00}}};
+
+static int ili9806e_write_reg(const struct device *dev, uint8_t reg, const uint8_t *buf, size_t len)
+{
+	int ret;
+	const struct ili9806e_config *cfg = dev->config;
+
+	ret = mipi_dsi_dcs_write(cfg->mipi_dsi, cfg->channel, reg, buf, len);
+	if (ret < 0) {
+		LOG_ERR("Failed writing reg: 0x%x result: (%d)", reg, ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int ili9806e_write_reg_val(const struct device *dev, uint8_t reg, uint8_t value)
+{
+	return ili9806e_write_reg(dev, reg, &value, 1);
+}
+
+static int ili9806e_write_sequence(const struct device *dev, const struct ili9806e_init_cmd *cmd,
+				   uint8_t nr_cmds)
+{
+	int ret = 0;
+
+	/* Loop trough all commands as long as writes are successful*/
+	for (int i = 0; i < nr_cmds && ret == 0; i++) {
+		ret = ili9806e_write_reg(dev, cmd->reg, cmd->cmd, cmd->cmd_len);
+		cmd++;
+	}
+
+	return ret;
+}
+
+static int ili9806e_config(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	ret = ili9806e_write_sequence(dev, init_cmds, ARRAY_SIZE(init_cmds));
+	if (ret < 0) {
+		return ret;
+	}
+	/* Add a delay, otherwise MADCTL not taken */
+	k_msleep(120);
+
+	/* Exit sleep mode */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Wait for sleep out exit */
+	k_msleep(5);
+
+	/* Set color mode */
+	ret = ili9806e_write_reg_val(dev, MIPI_DCS_SET_PIXEL_FORMAT,
+				     cfg->pixel_format == PIXEL_FORMAT_RGB_565
+					     ? ILITEK_ILI9806E_COLMOD_RGB565
+					     : ILITEK_ILI9806E_COLMOD_RGB888);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Turn on display */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	return ret;
+}
+
+static int ili9806e_blanking_on(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int ili9806e_blanking_off(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int ili9806e_write(const struct device *dev, uint16_t x, uint16_t y,
```

Comment:
Why implement this function? If it is absent, the display driver API will still return an error- is this present simply for the error log?

Reply:
Done. I removed this function